common-library/src/main/java/com/sdevelopment/skt/common/domain/Product.java

```diff
@@ -0,0 +1,89 @@
+package com.sdevelopment.skt.common.domain;
+
+import org.hibernate.validator.constraints.NotBlank;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.NamedStoredProcedureQueries;
+import javax.persistence.NamedStoredProcedureQuery;
+import javax.persistence.ParameterMode;
+import javax.persistence.StoredProcedureParameter;
+import javax.persistence.Table;
+import java.io.Serializable;
+
+@Entity
+@Table(name = "products")
+@NamedStoredProcedureQueries({
+        @NamedStoredProcedureQuery(name = "saveProduct",
+                procedureName = "InsertProduct",
+                parameters = {
+                        @StoredProcedureParameter(mode = ParameterMode.IN, name = "pname", type = String.class)
+                }),
+
+        @NamedStoredProcedureQuery(name = "getAllProducts",
+                procedureName = "GetAllProducts",
+                resultClasses = Product.class)
+})
+public class Product implements Serializable {
+    @Id
+    private Integer id;
+
+    @NotBlank(message = "Name may not be empty")
+    private String name;
+
+    public Product() {
+
+    }
+
+    public Product(Integer id, String name) {
+        this.id = id;
+        this.name = name;
+    }
+
+    public Integer getId() {
+        return id;
+    }
+
+    public void setId(Integer id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
there are better ways to write `equals` and `hashCode` methods, please check items 8 and 9 of the Effective Java book. You can even use IDE to autogenerate them.

Reply:
Implemented as per Effective Java and also created a unit test.